src/main/target/SITL/sitl.c

```diff
@@ -467,7 +550,15 @@ static bool pwmEnableMotors(void)
 
 static void pwmWriteMotor(uint8_t index, float value)
 {
-    motorsPwm[index] = value - idlePulse;
+    if (pthread_mutex_trylock(&updateLock) != 0) return;
+    if (index < 4) {
```

Comment:
Betaflight supports up to 8 motors. If this specifically needs to be limited to only 4 motors in Gazebo (for whatever reason), let me know.
```suggestion
    if (index < MAX_SUPPORTED_MOTORS) {
```